keyless/pepper/service/src/lib.rs

```diff
@@ -191,24 +215,39 @@ fn process_common(
         "PepperInput is available."
     );
     let input_bytes = bcs::to_bytes(&input).unwrap();
-    let (pepper, vuf_proof) = vuf::bls12381_g1_bls::Bls12381G1Bls::eval(&VUF_SK, &input_bytes)
+    let (pepper_base, vuf_proof) = vuf::bls12381_g1_bls::Bls12381G1Bls::eval(&VUF_SK, &input_bytes)
         .map_err(|e| InternalError(format!("bls12381_g1_bls eval error: {e}")))?;
     if !vuf_proof.is_empty() {
         return Err(InternalError("proof size should be 0".to_string()));
     }
 
+    let pinkas_pepper = PinkasPepper::from_affine_bytes(&pepper_base)
+        .or_else(|_| Err(InternalError("Failed to derive pinkas pepper".to_string())))?;
+    let master_pepper = pinkas_pepper.to_master_pepper();
+    let derived_pepper = ExtendedPepper::from_seed(master_pepper.to_bytes())
+        .or_else(|e| Err(InternalError(e.to_string())))?
+        .derive(&checked_derivation_path)
+        .or_else(|e| Err(InternalError(e.to_string())))?;
+
     if encrypts_pepper {
```

Comment:
Hopefully, we can remove this `if` :)

Reply:
since this comment, we've concluded encryption is not needed.